mlflow/store/artifact/databricks_artifact_repo.py

```diff
@@ -231,6 +237,68 @@ def _azure_upload_file(self, credentials, local_file, artifact_path):
         except Exception as err:
             raise MlflowException(err)
 
+    def _azure_adls_gen2_upload_file(self, credentials, local_file, artifact_path):
+        """
+        Uploads a file to a given Azure storage location using the ADLS gen2 API.
+        """
+        try:
+            headers = self._extract_headers_from_credentials(credentials.headers)
+
+            # try to create the file
+            try:
+                put_adls_file_creation(credentials.signed_uri, headers=headers)
+            except requests.HTTPError as e:
+                if e.response.status_code in [401, 403]:
+                    _logger.info(
+                        "Failed to authorize ADLS file creation request, possibly due "
+                        "to credential expiration. Refreshing credentials and trying again..."
+                    )
+                    credential_info = self._get_write_credential_infos(
```

Comment:
@kriscon-db Shall we just assign it to "credentials" so that we are using the correct signed url in later patch and flush? Same for patch code block.

Reply:
good call.  yes, please feel free to change this.